test/libsolidity/syntaxTests/abiEncoder/abi_encode_call_tuple_from_function.sol

```diff
@@ -0,0 +1,15 @@
+contract C {
+    function g0() internal {}
+    function g1() internal returns (uint) {}
+    function g2() internal returns (uint, uint) {}
+
+    function f() public pure {
+        abi.encodeCall(this.f, g0());
+        abi.encodeCall(this.f, g1());
+        abi.encodeCall(this.f, g2());
```

Comment:
Also, please add this one too unless we already have a test for it:
```
        abi.encodeCall(this.f, (g0()));
        abi.encodeCall(this.f, (g1()));
        abi.encodeCall(this.f, (g2()));
```

It will be a different error and does not cause `bad_cast` but we should just have it covered.